### PR TITLE
Command level logging changes

### LINKED ISSUE #741: Command-level logging
Let's add logging at the DEBUG level for each command. 

This should contain the name of the command as already used in the CommandProcessor metrics, for example: `FindCommand`, `DeleteManyCommand`, etc.

The logging should also include pertinent data per call such as:

- tenant ID, namespace, collection name
- size of incoming/outgoing document(s)
- for `insertMany`, `updateMany`, `deleteMany`, the number of documents in the request, and the number of documents handled successfully.
- the duration of the request (maybe, because it is already captured in metrics)

### PR DESCRIPTION
**Pull request by kathirsvn**
**What this PR does**:
Added a new feature to log command level details when enabled.

**Which issue(s) this PR fixes**:
Fixes #741 

**Checklist**
- [X] Changes manually tested
- [X] Automated Tests added/updated
- [X] Documentation added/updated
- [X] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/config/CommandLevelLoggingConfig.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  /** @return If request info logging is enabled. */
+  @WithDefault("false")
+  boolean enabled();
+
+  /** @return If only requests with errors should be logged. */
+  @WithDefault("true")
+  boolean onlyResultsWithErrors();
+
+  /** @return Set of tenants for which the request info should be logged. */
+  @WithDefault(ALL_TENANTS)
```
[the remark this answers is not in the document]
Comment (jeffreyscarpenter) [reply src/main/java/io/stargate/sgv2/jsonapi/config/CommandLevelLoggingConfig.java:40]: Can't you set it with something like this?

`-Dstargate.jsonapi.logging.enabled-tenants="id1,id2,id3"`

**Review by jeffreyscarpenter — APPROVED** [review]
Looks great to me

File: src/main/java/io/stargate/sgv2/jsonapi/config/CommandLevelLoggingConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/config/CommandLevelLoggingConfig.java:40]: @kathirsvn My comment was more wrt usability of this feature and repeated string comparison to print log. If this feature is needed, we are good to merge the PR.

[the remark this answers is not in the document]
Comment (kathirsvn) [reply src/main/java/io/stargate/sgv2/jsonapi/config/CommandLevelLoggingConfig.java:40]: - By default, this feature when enabled will add this command level log for all tenants which is the default "ALL" tenant. 
- Tenants can be specified comma separated as @jeffreyscarpenter mentioned above as well.
- O(1) contains check is done in the given tenant's list when command level logging is enabled, so should be good.

[the remark this answers is not in the document]
Comment (kathirsvn) [reply src/main/java/io/stargate/sgv2/jsonapi/config/CommandLevelLoggingConfig.java:40]: > We will want to add usage of these options to the `CONFIGURATION.md`

Sure, makes sense, ll add a section.

[the remark this answers is not in the document]
Comment (kathirsvn) [reply src/main/java/io/stargate/sgv2/jsonapi/config/CommandLevelLoggingConfig.java:40]: Updated CONFIGURATION.md in [29db0e6](https://github.com/stargate/jsonapi/pull/794/commits/29db0e6ecaf73e5f5e44772d4f22dd2ac8d32952).

File: src/main/java/io/stargate/sgv2/jsonapi/service/processor/MeteredCommandProcessor.java [hunk 2]
```diff
@@ -93,9 +107,98 @@ public <T extends Command> Uni<CommandResult> processCommand(
...
+    CommandLog commandLog =
+        new CommandLog(
+            command.getClass().getSimpleName(),
+            stargateRequestInfo.getTenantId().orElse(UNKNOWN_VALUE),
+            commandContext.namespace(),
+            commandContext.collection(),
+            getIncomingDocumentsCount(command),
+            getOutgoingDocumentsCount(result),
+            result != null ? result.errors() : Collections.emptyList());
+    return new ObjectMapper().valueToTree(commandLog).toString();
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/processor/MeteredCommandProcessor.java:143]: Ughhhh. This must be changed to reuse `ObjectMapper`...

**Pull request merged**